some files in pkg/ are downloaded by multiple packages

```
$ make -j 7
...
[download]  ocaml-native                                                                                 
[download]  ocaml-flexdll                                                                                
[download]  ocaml-core                                                                                   

Download failed or wrong checksum of package ocaml-native!
------------------------------------------------------------
  3000K .......... .......... .......... .......... .......... 91% 9.79M 0s
  3050K .......... .......... .......... .......... .......... 93% 11.4M 0s
  3100K .......... .......... .......... .......... .......... 95% 12.9M 0s
  3150K .......... .......... .......... .......... .......... 96% 11.0M 0s
  3200K .......... .......... .......... .......... .......... 98% 11.4M 0s
  3250K .......... .......... .......... .......... .......... 99% 11.3M 0s
  3300K .......... .....                                      100% 8.04M=0.4s
2015-12-16 13:27:28 (8.26 MB/s) - written to stdout [3394854/3394854]
------------------------------------------------------------
[log]      /home/mxe/mxe-libmysqlclient-++-plus-plus/log/ocaml-native-download

Makefile:423: recipe for target 'download-only-ocaml-native' failed
make: *** [download-only-ocaml-native] Error 1
make: *** Waiting for unfinished jobs....
```

Some packages download same pkg files. It causes errors (see above). I have two ideas howto fix these errors:

  * remove `$(PKG)_URL` from dependent package. E.g. from ocaml-core. Add automatic MXEBot check that all `$(PKG)_URL`s are unique.
  * or lock downloaded file using a mutes. A [mkdir based mutex](http://wiki.bash-hackers.org/howto/mutex) can be used.

--- a/Makefile
+++ b/Makefile
@@ -426,7 +426,12 @@ define PKG_RULE
 download-$(1): $(addprefix download-,$($(1)_DEPS)) download-only-$(1)
 
 .PHONY: download-only-$(1)
-download-only-$(1):
+# Packages can share a source archive to build different sets of features
+# or dependencies (see bfd/binutils openscenegraph/openthreads qwt/qwt_qt4).
+# Use a double-colon rule to allow multiple definitions:
+# https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html
+download-only-$(1): download-only-$($(1)_FILE)
+download-only-$($(1)_FILE)::
 	$(and $($(1)_URL),
 	@[ -d '$(LOG_DIR)/$(TIMESTAMP)' ] || mkdir -p '$(LOG_DIR)/$(TIMESTAMP)'
 	@if ! $(call CHECK_PKG_ARCHIVE,$(1)); then \

